mibidata/tiff.py

```diff
@@ -343,26 +342,44 @@ def _convert_from_previous(description):
     if not description.get('mibi.fov_name') and description.get(
             'mibi.description'):
         description['mibi.fov_name'] = description.pop('mibi.description')
-    # TODO: Clean up repetition between this and the same MibiImage method
-    if description.get('mibi.folder') and not description.get('mibi.fov_id'):
-        description['mibi.fov_id'] = description['mibi.folder'].split('/')[0]
-        warnings.warn(
-            'The "fov_id" attribute is now required if "folder" is '
-            'specified. Setting "fov_id" to {}.'.format(
-                description['mibi.fov_id']))
-    if (not description.get('mibi.folder') and description.get('mibi.fov_id')
-            and description.get('mibi.fov_id').startswith('FOV')):
-        description['mibi.folder'] = description['mibi.fov_id']
-        warnings.warn(
-            'The "folder" attribute is required if "fov_id" is specified. '
-            'Setting "folder" to {}.'.format(description['mibi.folder']))
+    if description.get('mibi.folder'):
+        description['mibi.fov_id'] = ''
+        value = 'mibi.folder'
+        field = 'mibi.fov_id'
+        value_name = 'folder'
+        field_name = 'fov_id'
+        matchNeeded = True
+    if description.get('mibi.fov_id'):
+        description['mibi.folder'] = ''
+        value = 'mibi.fov_id'
+        field = 'mibi.folder'
+        value_name = 'fov_id'
```

Comment:
> I would suggest filling field_name and value_name by doing value.split('.')[1] rather than assume the length of the prefix. Though, can you just set value and field above to folder or fov_id without the mibi prefix?

Sorry, I see now that `value` and `field` need to have the `'mibi.'` prefix, but rather than define new variables here, below when you call `mi.MibiImage.match_fov_folder()` use `value.split('.')[1]`.